### PR TITLE
Verify server signature

### PR DESCRIPTION
**Pull request by tohar-ayash**

### CODE REVIEW

**Comment by gennadylaventman** [comment]
I think you need unit test for that.

**Review by gennadylaventman — COMMENTED** [review]
Lets add unit test here.

**Comment by tock-ibm** [comment]
Look for the TODO comments in the code, many of them have `TODO signature verification` 
address these places in the code where you find these.

I think also there needs to be a testcase on every context type (data, user, admin, config, json-query, provenance, etc) that tests signature verification, i.e. it fails if the signature is wrong.

File: pkg/bcdb/tx_context_test.go [hunk 1]
```diff
@@ -619,6 +644,93 @@ func TestTxQuery(t *testing.T) {
 
 }
 
+func TestResponseSelector(t *testing.T) {
+	emptySigner := &mocks.Signer{}
```
Comment (gennadylaventman) [inline pkg/bcdb/tx_context_test.go:648]: This is overkill. You create all this mock infrastructure, but actually don't use it in test. Just calling to `ResponseSelector` twice, one with correct envelope and one with in-correct, will be enough.

**Review by gennadylaventman — CHANGES_REQUESTED** [review]

File: pkg/bcdb/tx_context_test.go (hunk 1 above)
Comment (tohar-ayash) [reply pkg/bcdb/tx_context_test.go]: Done

**Review by gennadylaventman — APPROVED** [review]

**Pull request merged**